DataManagementSystem/DB/FileCatalogComponents/MultiVOMetaNameMixIn.py

```diff
@@ -0,0 +1,51 @@
+""" DIRAC Multi VO MixIn class to manage file metadata and directory for multiple VO.
+"""
+
+__RCSID__ = "$Id$"
+
+from DIRAC.DataManagementSystem.DB.FileCatalogComponents.MetaNameMixIn import MetaNameMixIn
+from DIRAC.ConfigurationSystem.Client.Helpers import Registry
+
+
+class MultiVOMetaNameMixIn(MetaNameMixIn):
+  """
+  MULti-VO MetaName MixIn implementation.
+  """
+
+  def __init__(self, *args, **kwargs):
+    super(MultiVOMetaNameMixIn, self).__init__(*args, **kwargs)
+
+  def getMetaName(self, meta, credDict):
+    """
+    Return a fully-qualified metadata name based on client-suplied metadata name and
+    client credentials. User group is added to the metadata passed in.
+
+    :param meta: metadata name
+    :param credDict: client credentials
+    :return: fully-qualified metadata name
+    """
+
+    return meta + self.getMetaNameSuffix(credDict)
+
+  def getMetaNameSuffix(self, credDict):
+    """
+    Get a VO specific suffix from user credentials.
+
+    :param credDict: user credentials
+    :return: VO specific suffix
+    """
+    vo = Registry.getGroupOption(credDict['group'], 'VO')
+    return '_' + vo.replace('-', '_')
+
+  def stripSuffix(self, metaDict, credDict):
+    """
+    Strip the suffix from all keys which contain it, leaving other keys intact.
+    :param metaDict: original dict
```

Comment:
Need a newline before the first `:param ...`

Reply:
done